File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.cpp
class MyStack {
public: 
    queue<int> q1,q2;
    MyStack() {
        
    }
    
    void push(int x) {
        // q1 []  q2 [] // 
        // push q1] 
        q1.push(x); 
        while(q2.size()){
            q1.push(q2.front());
            q2.pop();
        }

        //q2 
        while(q1.size()){
            q2.push(q1.front());
            q1.pop();
        }

    }
    
    int pop() {
        int front = q2.front();
        q2.pop();
        return front;
    }
    
    int top() {
        return q2.front();
    }
    
    bool empty() {
        return q2.empty();
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
from collections import deque 

class MyStack:
    
    def __init__(self):
        self.q = deque()    

    def push(self, x: int) -> None:
        copy = deque()
        while self.q:
            copy.append(self.q.popleft())
        self.q.append(x) 
        while copy: 
            self.q.append(copy.popleft())

    def pop(self) -> int:
        return self.q.popleft()

    def top(self) -> int:
        return self.q[0]

    def empty(self) -> bool:
        return len(self.q)==0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()